helpers/userProfile.db.ts

```diff
@@ -27,7 +27,7 @@ export const userNameExists = async (username: string) => {
 export const userProfileExists = async (walletAddress: string) => {
   try {
     await dbConnect()
-    const doc = await UserProfile.countDocuments({ walletAddress })
+    const doc = await UserProfile.countDocuments({ walletAddress: walletAddress?.toLowerCase() })
```

Comment:
you can do something like this if you want to ignore case
`await UserProfile.countDocuments({walletAddress:{ $regex: walletAddress, $options: "i" }})`